src/main/scala/mesosphere/marathon/core/instance/Reservation.scala

```diff
@@ -63,8 +63,7 @@ object Reservation {
     def apply(label: String): Id = label match {
       case SimplifiedIdRegex(safeRunSpecId, prefix, uuid) =>
         val runSpec = PathId.fromSafePath(safeRunSpecId)
-        val instanceId = Instance.Id(runSpec, Prefix.fromString(prefix), UUID.fromString(uuid))
-        SimplifiedId(instanceId)
+        LegacyId(runSpec, "", UUID.fromString(uuid))
```

Comment:
I'm not sure this is right. This is basically the factory method. When we match the `SimplifiedId` we should not return the `LegacyId`. What do you do if the reservation id is a `SimplifiedId`?

Reply:
yeah I don't think this is right either... That was pre-lunch thing and I touched different part of code than I wanted to :) now it should be good... that's the part of code that `InstanceOpFactoryImpl` uses